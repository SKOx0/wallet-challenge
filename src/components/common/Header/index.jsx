import React from 'react';
import { Navbar, Container, NavbarBrand, NavbarItem, NavbarMenu, NavbarEnd, NavbarStart } from 'bloomer';
// import FontAwesomeIcon from '@fortawesome/react-fontawesome';
// import * as Icons from '@fortawesome/fontawesome-free-solid';

const Header = () => (
  <Navbar className="has-shadow is-spaced">
    <Container>
      <NavbarBrand>
        <NavbarItem>
          Wallet Rock
        </NavbarItem>
      </NavbarBrand>

      <NavbarMenu>
        <NavbarStart>
          <NavbarItem href="#" isHidden="desktop">
            {/* <FontAwesomeIcon icon={Icons.faSignOutAlt} className="mr-5"></FontAwesomeIcon> */}
            <span>Sair</span>
          </NavbarItem>
        </NavbarStart>
        <NavbarEnd>
          <NavbarItem href="#" isHidden="touch">
            {/* <FontAwesomeIcon icon={Icons.faSignOutAlt} className="mr-5"></FontAwesomeIcon> */}
            <span>Sair</span>
          </NavbarItem>
        </NavbarEnd>
      </NavbarMenu>
    </Container>
  </Navbar>
);

export default Header;
