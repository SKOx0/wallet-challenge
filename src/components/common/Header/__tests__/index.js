import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../index';

describe('<Header />', () => {
  it('deve ser igual ao snapshot, com configurações padrões', () => {
    const renderedComponent = shallow(<Header onLogout={() => {}} />);

    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
