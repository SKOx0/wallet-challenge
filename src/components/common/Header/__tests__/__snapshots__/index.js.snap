// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> deve ser igual ao snapshot, com configurações padrões 1`] = `
<withHelpersModifiers(Navbar)
  className="has-shadow is-spaced"
>
  <withHelpersModifiers(Container)>
    <withHelpersModifiers(NavbarBrand)>
      <withHelpersModifiers(NavbarItem)>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
           Wallet Rock 
        </NavLink>
      </withHelpersModifiers(NavbarItem)>
      <withHelpersModifiers(NavbarItem)
        href="#"
        isHidden="desktop"
      >
        <FontAwesomeIcon
          border={false}
          className="mr-5"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f2f5",
                "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
              ],
              "iconName": "sign-out-alt",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
        <span>
          Sair
        </span>
      </withHelpersModifiers(NavbarItem)>
    </withHelpersModifiers(NavbarBrand)>
    <withHelpersModifiers(NavbarMenu)>
      <withHelpersModifiers(NavbarEnd)>
        <withHelpersModifiers(NavbarItem)
          isHidden="touch"
          onClick={[Function]}
        >
          <withHelpersModifiers(Button)>
            <FontAwesomeIcon
              border={false}
              className="mr-5"
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f2f5",
                    "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                  ],
                  "iconName": "sign-out-alt",
                  "prefix": "fas",
                }
              }
              listItem={false}
              mask={null}
              name=""
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              transform={null}
            />
            <span>
              Sair
            </span>
          </withHelpersModifiers(Button)>
        </withHelpersModifiers(NavbarItem)>
      </withHelpersModifiers(NavbarEnd)>
    </withHelpersModifiers(NavbarMenu)>
  </withHelpersModifiers(Container)>
</withHelpersModifiers(Navbar)>
`;
